### ISSUE TITLE
proposal: notify when the request.Context is not used

### ISSUE THREAD

**Issue by komuw** [kkHAIKE/contextcheck#5]
Go is frequently used for servers. And as such http.Handler is used often. 

Given the following code;
```go
package main

import (
	"context"
	"net/http"
)
func foo(ctx context.Context, a, b int) {}

func myHandler(w http.ResponseWriter, r *http.Request) {
	foo(context.Background(), 2, 33)
}

func main() {}
```

I think `contextcheck` should error with a message like;
`Non-inherited new context, use `r.Context()` instead`

And to resolve the error, you would have to do:
```
foo(r.Context(), 2, 33)
```

**Comment by kkHAIKE** [comment]
duplicate https://github.com/sylvia7788/contextcheck/issues/3

**Comment by kkHAIKE** [comment]
good idea 😊

**Comment by kkHAIKE** [comment]
Done in new version, https://github.com/golangci/golangci-lint/pull/3102
